guides/hack/16-readonly/01-introduction.md

```diff
@@ -0,0 +1,49 @@
+`readonly` is a keyword used to create immutable references to [Objects](https://docs.hhvm.com/hack/classes/introduction) and their properties. 
+
+### How does it work?
+Expressions in Hack can be annotated with the `readonly` keyword. When an object or reference is readonly, there are two main constraints on it:
+* **Readonlyness:** Object properties can not be modified (i.e. mutated).
+* **Deepness:** All nested properties of a readonly value are readonly.
+
+### Readonlyness
+Object properties of readonly values can not be modified (i.e. mutated).
+
+``` Hack
```

Comment:
This is cool! qq: this means that each test file would need to be self contained, though, right? Which means adding a lot of class definitions that will increase the verbosity of the examples

That said, I'm all for this since it would make each example runnable on its own

Reply:
- If they're sharable between examples, add a `.inc.hack` file with them
- verbosity: I'm not seeing any usage of this, but it looks like if you add `// @example-start` and `// @example-end` comments, only the code between them will be rendered